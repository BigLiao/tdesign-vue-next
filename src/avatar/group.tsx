import { defineComponent } from 'vue';
import { prefix } from '../config';
import props from './avatar-group-props';
import { TNodeReturnValue } from '../common';
import Avatar from './avatar';
import { renderContent, renderTNodeJSX } from '../utils/render-tnode';

const name = `${prefix}-avatar-group`;

export default defineComponent({
  name: 'TAvatarGroup',
  components: {
    Avatar,
  },

  provide(): Record<string, any> {
    return {
      avatarGroup: this,
    };
  },

  props: {
    ...props,
  },

  methods: {
    renderEllipsisAvatar(children: Array<TNodeReturnValue>): Array<TNodeReturnValue> {
      if (children?.length > this.max) {
        const content = this.setEllipsisContent(children);
        const outAvatar = children.slice(0, this.max);
        outAvatar.push(<Avatar size={this.size} icon={this.isIcon() && typeof this.collapseAvatar !== 'string' ? this.collapseAvatar : null}>{content}</Avatar>);
        return [outAvatar];
      }
      return [children];
    },
    setEllipsisContent(children: Array<TNodeReturnValue>) {
      let content: any = '';
      if (this.collapseAvatar) {
        if (!this.isIcon()) {
          content = renderContent(this, 'collapseAvatar', 'content');
        }
      } else {
        content = `+${children.length - this.max}`;
      }
      return content;
    },
    isIcon() {
      const content = renderTNodeJSX(this, 'collapseAvatar');
      return content;
    },
  },
  render() {
    const { $slots } = this;
    const children: TNodeReturnValue = $slots.default && $slots.default(null);
    const { cascading, max } = this.$props;
    const groupClass = [
      `${name}`,
      {
        [`${prefix}-avatar--offset-right`]: cascading === 'right-up',
        [`${prefix}-avatar--offset-left`]: cascading === 'left-up',
      },
    ];
    let content = [children];

    if (max && max >= 0) {
      content = [this.renderEllipsisAvatar(children)];
    }
    return <div class={groupClass}>
      {content}
    </div>;
  },

});
