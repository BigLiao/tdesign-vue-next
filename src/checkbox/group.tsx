import { defineComponent, h } from 'vue';
import { prefix } from '../config';
import Checkbox from './checkbox';
import checkboxGroupProps from '@TdTypes/checkbox-group/props';
import { CheckboxOptionObj, TdCheckboxProps, CheckboxGroupValue } from '@TdTypes/checkbox/TdCheckboxProps';
import intersection from 'lodash/intersection';

const name = `${prefix}-checkbox-group`;

export default defineComponent({
  name,
  components: {
    Checkbox,
  },
  emits: ['change'],
  props: { ...checkboxGroupProps },

  data() {
    return {
      checkedMap: {},
    };
  },

  watch: {
    values: {
      immediate: true,
      handler() {
        if (this.value instanceof Array) {
          const map = {};
          this.value.forEach((item: string | number) => {
            map[item] = true;
          });
          this.checkedMap = map;
        }
      },
    },
  },

  computed: {
    optionList(): Array<CheckboxOptionObj> {
      if (!this.options) return [];
      return this.options.map((item) => {
        let r: CheckboxOptionObj = {};
        if (typeof item !== 'object') {
          r = { label: String(item), value: item } ;
        } else {
          r = { ...item };
          r.disabled = r.disabled === undefined ? this.disabled : r.disabled;
        }
        return r;
      });
    },
    values(): string {
      if (this.value instanceof Array) {
        return this.value.join();
      }
      return '';
    },
    intersectionLen(): number {
      const values = this.optionList.map(item => item.value);
      if (this.value instanceof Array) {
        const n = intersection(this.value, values);
        return n.length;
      }
      return 0;
    },
    isCheckAll(): boolean {
      if (this.value instanceof Array && this.value.length !== this.optionList.length - 1) {
        return false;
      }
      return this.intersectionLen === this.optionList.length - 1;
    },
    indeterminate(): boolean {
      return !this.isCheckAll && this.intersectionLen < this.optionList.length && this.intersectionLen !== 0;
    },
  },

  render() {
    return (
      <div class={name}>
        {!!this.optionList.length && this.optionList.map((option, index) => {
          if (option.checkAll) return this.renderCheckAll(option);
          return (
            <Checkbox
              key={index}
              props={{ ...option }}
              checked={this.checkedMap[option.value]}
            >
              {this.renderLabel(option)}
            </Checkbox>
          );
        })}
        {this.$slots.default() && this.$slots.default(null)}
      </div>
    );
  },

  methods: {
    renderCheckAll(option: CheckboxOptionObj) {
      return (
        <Checkbox
          checked={this.isCheckAll}
          indeterminate={this.indeterminate}
          onChange={this.onCheckAllChange}
          data-name='TDESIGN_CHECK_ALL'
          props={{ ...option }}
        >{this.renderLabel(option)}</Checkbox>
      );
    },
    renderLabel(option: CheckboxOptionObj) {
      if (typeof option.label === 'function') {
        return option.label(h);
      }
      return option.label;
    },
    emitChange(val: CheckboxGroupValue, e?: Event) {
      this.$emit('change', val, { e });
    },
    handleCheckboxChange(data: { checked: boolean; e: Event; option: TdCheckboxProps }) {
      const oValue = data.option.value;
      if (this.value instanceof Array) {
        const val = [...this.value];
        if (data.checked) {
          val.push(oValue);
        } else {
          const i = val.indexOf(oValue);
          val.splice(i, 1);
        }
        this.emitChange(val, data.e);
        this.setCheckedMap(data.option.value, data.checked);
      }
    },
    setCheckedMap(value: string | number, checked: boolean) {
      this.checkedMap[value] = checked;
    },
    onCheckAllChange(checked: boolean, context: { e: Event }) {
      if (checked) {
        const val = [];
        for (let i = 0, len = this.optionList.length; i < len; i++) {
          const item = this.optionList[i];
          if (item.checkAll) continue;
          val.push(item.value);
          this.checkedMap[item.value] = true;
        }
        this.emitChange(val, context.e);
      } else {
        this.emitChange([], context.e);
        this.checkedMap = {};
      }
    },
  },
});