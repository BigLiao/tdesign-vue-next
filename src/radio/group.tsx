import { VNode, defineComponent, h } from 'vue';
import isString from 'lodash/isString';
import isNumber from 'lodash/isNumber';
import props from '../../types/radio-group/props';
import { RadioOptionObj, RadioOption, RadioValue } from '../../types/radio/TdRadioProps';
import { prefix } from '../config';
import Radio from './radio';
import { TNodeReturnValue } from '../common';

const name = `${prefix}-radio-group`;

export default defineComponent({
  name,
  components: {
    Radio,
  },
  provide() {
    return {
      radioGroup: this,
    };
  },
  props: { ...props },
  emits: ['change'],
  methods: {
    handleRadioChange(value: RadioValue, context: { e: Event }) {
      this.$emit('change', value, context);
    },
  },
  render(): VNode {
    const { $slots } = this;
    let children: TNodeReturnValue = $slots.default && $slots.default(null);

    if (this.options && this.options.length) {
      children = (this.options).map((option: RadioOption) => {
        let opt = option as RadioOptionObj;
        if (isNumber(option) || isString(option)) {
          opt = { value: option, label: option.toString() };
        }
        return (
          <Radio
            key={`radio-group-options-${opt.value}-${Math.random()}`}
            name={this.name}
            checked={this.value === opt.value}
            disabled={'disabled' in opt ? opt.disabled : this.disabled}
            value={opt.value}
          >
            {typeof opt.label === 'function' ? opt.label(h) : opt.label}
          </Radio>
        );
      });
    }

    const groupClass = [
      `${name}`,
      `${name}-${this.buttonStyle}`,
      `${name}-${this.size}`,
    ];

    return (
      <div class={groupClass}>
        {children}
      </div>
    );
  },
});
