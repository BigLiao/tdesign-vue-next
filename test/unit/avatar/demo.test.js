import { mount } from '@vue/test-utils';
import base from '@/examples/avatar/demos/base.vue';
import shape from '@/examples/avatar/demos/shape.vue';
import size from '@/examples/avatar/demos/size.vue';
import adjust from '@/examples/avatar/demos/adjust.vue';
import group from '@/examples/avatar/demos/group.vue';
import groupMax from '@/examples/avatar/demos/group-max.vue';
import groupCascading from '@/examples/avatar/demos/group-cascading.vue';

// unit test for component in examples.
describe('Avatar', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Avatar', () => {
  it('shape demo works fine', () => {
    const wrapper = mount(shape);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Avatar', () => {
  it('size demo works fine', () => {
    const wrapper = mount(size);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Avatar', () => {
  it('size demo works fine', () => {
    const wrapper = mount(adjust);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Avatar', () => {
  it('group demo works fine', () => {
    const wrapper = mount(group);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Avatar', () => {
  it('groupMax demo works fine', () => {
    const wrapper = mount(groupMax);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Avatar', () => {
  it('groupCascading demo works fine', () => {
    const wrapper = mount(groupCascading);
    expect(wrapper.element).toMatchSnapshot();
  });
});
