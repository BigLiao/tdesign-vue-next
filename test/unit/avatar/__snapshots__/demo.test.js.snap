// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar base demo works fine 1`] = `
<div
  class="demo-avatar"
>
  <div
    class="t-avatar t-avatar--circle"
  >
    <img
      alt=""
      src="https://tdesign.gtimg.com/site/avatar.jpg"
    />
  </div>
  <div
    class="t-avatar t-avatar--circle"
  >
    <span>
      
      W
      
    </span>
  </div>
  <div
    class="t-avatar t-avatar--circle t-avatar__icon"
  >
    
    <svg
      class="t-icon t-icon-user"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 576C601.92257344 576 686.4307968 599.18159296 759.8745728 639.8946848L760.64 640 797.2160704 658.3197248C818.5708096 669.2764608 832 691.2604224 832 715.261984L832 832C832 867.3462208 803.3462208 896 768 896L256 896C220.653776 896 192 867.3462208 192 832L192 715.2818624C192 691.2790592 205.43055936 669.2941888 226.7869856 658.3380288L263.744 640 264.064 639.936 273.52287808 634.81298816C344.74231744 597.2582336 425.8885952 576 512 576ZM512 640C416.8213984 640 328.5751328 669.6808576 256.01207488 720.2916928L256 832 768 832 768.0210368 720.3147904C695.4517184 669.689888 607.19308032 640 512 640ZM512 159.72895488C618.038672 159.72895488 704 245.69028288 704 351.72895488 704 457.76762624 618.038672 543.72895488 512 543.72895488 405.961328 543.72895488 320 457.76762624 320 351.72895488 320 245.69028288 405.961328 159.72895488 512 159.72895488ZM512 223.72895488C441.307552 223.72895488 384 281.03650688 384 351.72895488 384 422.42140224 441.307552 479.72895488 512 479.72895488 582.692448 479.72895488 640 422.42140224 640 351.72895488 640 281.03650688 582.692448 223.72895488 512 223.72895488Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    
  </div>
  <div
    class="t-avatar t-avatar--circle t-avatar__icon"
  >
    
    
    <svg
      class="t-icon t-icon-user"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 576C601.92257344 576 686.4307968 599.18159296 759.8745728 639.8946848L760.64 640 797.2160704 658.3197248C818.5708096 669.2764608 832 691.2604224 832 715.261984L832 832C832 867.3462208 803.3462208 896 768 896L256 896C220.653776 896 192 867.3462208 192 832L192 715.2818624C192 691.2790592 205.43055936 669.2941888 226.7869856 658.3380288L263.744 640 264.064 639.936 273.52287808 634.81298816C344.74231744 597.2582336 425.8885952 576 512 576ZM512 640C416.8213984 640 328.5751328 669.6808576 256.01207488 720.2916928L256 832 768 832 768.0210368 720.3147904C695.4517184 669.689888 607.19308032 640 512 640ZM512 159.72895488C618.038672 159.72895488 704 245.69028288 704 351.72895488 704 457.76762624 618.038672 543.72895488 512 543.72895488 405.961328 543.72895488 320 457.76762624 320 351.72895488 320 245.69028288 405.961328 159.72895488 512 159.72895488ZM512 223.72895488C441.307552 223.72895488 384 281.03650688 384 351.72895488 384 422.42140224 441.307552 479.72895488 512 479.72895488 582.692448 479.72895488 640 422.42140224 640 351.72895488 640 281.03650688 582.692448 223.72895488 512 223.72895488Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    
    
    
  </div>
</div>
`;

exports[`Avatar group demo works fine 1`] = `
<div
  class="demo-avatar"
>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar-group t-avatar--offset-right"
    >
      
      
      <div
        class="t-avatar t-avatar--circle"
      >
        <img
          alt=""
          src="https://tdesign.gtimg.com/site/avatar.jpg"
        />
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          W
          
        </span>
      </div>
      <div
        class="t-avatar t-avatar--circle t-avatar__icon"
      >
        
        <svg
          class="t-icon t-icon-user"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 576C601.92257344 576 686.4307968 599.18159296 759.8745728 639.8946848L760.64 640 797.2160704 658.3197248C818.5708096 669.2764608 832 691.2604224 832 715.261984L832 832C832 867.3462208 803.3462208 896 768 896L256 896C220.653776 896 192 867.3462208 192 832L192 715.2818624C192 691.2790592 205.43055936 669.2941888 226.7869856 658.3380288L263.744 640 264.064 639.936 273.52287808 634.81298816C344.74231744 597.2582336 425.8885952 576 512 576ZM512 640C416.8213984 640 328.5751328 669.6808576 256.01207488 720.2916928L256 832 768 832 768.0210368 720.3147904C695.4517184 669.689888 607.19308032 640 512 640ZM512 159.72895488C618.038672 159.72895488 704 245.69028288 704 351.72895488 704 457.76762624 618.038672 543.72895488 512 543.72895488 405.961328 543.72895488 320 457.76762624 320 351.72895488 320 245.69028288 405.961328 159.72895488 512 159.72895488ZM512 223.72895488C441.307552 223.72895488 384 281.03650688 384 351.72895488 384 422.42140224 441.307552 479.72895488 512 479.72895488 582.692448 479.72895488 640 422.42140224 640 351.72895488 640 281.03650688 582.692448 223.72895488 512 223.72895488Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
        
        
      </div>
      
      
    </div>
  </div>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar-group t-avatar--offset-right"
    >
      
      
      <div
        class="t-avatar t-avatar--circle"
      >
        <img
          alt=""
          src="https://tdesign.gtimg.com/site/avatar.jpg"
        />
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          W
          
        </span>
      </div>
      <div
        class="t-avatar t-avatar--circle t-avatar__icon"
      >
        
        <svg
          class="t-icon t-icon-user"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 576C601.92257344 576 686.4307968 599.18159296 759.8745728 639.8946848L760.64 640 797.2160704 658.3197248C818.5708096 669.2764608 832 691.2604224 832 715.261984L832 832C832 867.3462208 803.3462208 896 768 896L256 896C220.653776 896 192 867.3462208 192 832L192 715.2818624C192 691.2790592 205.43055936 669.2941888 226.7869856 658.3380288L263.744 640 264.064 639.936 273.52287808 634.81298816C344.74231744 597.2582336 425.8885952 576 512 576ZM512 640C416.8213984 640 328.5751328 669.6808576 256.01207488 720.2916928L256 832 768 832 768.0210368 720.3147904C695.4517184 669.689888 607.19308032 640 512 640ZM512 159.72895488C618.038672 159.72895488 704 245.69028288 704 351.72895488 704 457.76762624 618.038672 543.72895488 512 543.72895488 405.961328 543.72895488 320 457.76762624 320 351.72895488 320 245.69028288 405.961328 159.72895488 512 159.72895488ZM512 223.72895488C441.307552 223.72895488 384 281.03650688 384 351.72895488 384 422.42140224 441.307552 479.72895488 512 479.72895488 582.692448 479.72895488 640 422.42140224 640 351.72895488 640 281.03650688 582.692448 223.72895488 512 223.72895488Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
        
        
      </div>
      
      
    </div>
  </div>
</div>
`;

exports[`Avatar groupCascading demo works fine 1`] = `
<div
  class="demo-avatar"
>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar-group t-avatar--offset-right"
    >
      
      
      <div
        class="t-avatar t-avatar--circle"
      >
        <img
          alt=""
          src="https://tdesign.gtimg.com/site/avatar.jpg"
        />
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          W
          
        </span>
      </div>
      <div
        class="t-avatar t-avatar--circle t-avatar__icon"
      >
        
        <svg
          class="t-icon t-icon-user"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 576C601.92257344 576 686.4307968 599.18159296 759.8745728 639.8946848L760.64 640 797.2160704 658.3197248C818.5708096 669.2764608 832 691.2604224 832 715.261984L832 832C832 867.3462208 803.3462208 896 768 896L256 896C220.653776 896 192 867.3462208 192 832L192 715.2818624C192 691.2790592 205.43055936 669.2941888 226.7869856 658.3380288L263.744 640 264.064 639.936 273.52287808 634.81298816C344.74231744 597.2582336 425.8885952 576 512 576ZM512 640C416.8213984 640 328.5751328 669.6808576 256.01207488 720.2916928L256 832 768 832 768.0210368 720.3147904C695.4517184 669.689888 607.19308032 640 512 640ZM512 159.72895488C618.038672 159.72895488 704 245.69028288 704 351.72895488 704 457.76762624 618.038672 543.72895488 512 543.72895488 405.961328 543.72895488 320 457.76762624 320 351.72895488 320 245.69028288 405.961328 159.72895488 512 159.72895488ZM512 223.72895488C441.307552 223.72895488 384 281.03650688 384 351.72895488 384 422.42140224 441.307552 479.72895488 512 479.72895488 582.692448 479.72895488 640 422.42140224 640 351.72895488 640 281.03650688 582.692448 223.72895488 512 223.72895488Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
        
        
      </div>
      
      
    </div>
  </div>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar-group t-avatar--offset-left"
    >
      
      
      <div
        class="t-avatar t-avatar--circle"
      >
        <img
          alt=""
          src="https://tdesign.gtimg.com/site/avatar.jpg"
        />
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          W
          
        </span>
      </div>
      <div
        class="t-avatar t-avatar--circle t-avatar__icon"
      >
        
        <svg
          class="t-icon t-icon-user"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 576C601.92257344 576 686.4307968 599.18159296 759.8745728 639.8946848L760.64 640 797.2160704 658.3197248C818.5708096 669.2764608 832 691.2604224 832 715.261984L832 832C832 867.3462208 803.3462208 896 768 896L256 896C220.653776 896 192 867.3462208 192 832L192 715.2818624C192 691.2790592 205.43055936 669.2941888 226.7869856 658.3380288L263.744 640 264.064 639.936 273.52287808 634.81298816C344.74231744 597.2582336 425.8885952 576 512 576ZM512 640C416.8213984 640 328.5751328 669.6808576 256.01207488 720.2916928L256 832 768 832 768.0210368 720.3147904C695.4517184 669.689888 607.19308032 640 512 640ZM512 159.72895488C618.038672 159.72895488 704 245.69028288 704 351.72895488 704 457.76762624 618.038672 543.72895488 512 543.72895488 405.961328 543.72895488 320 457.76762624 320 351.72895488 320 245.69028288 405.961328 159.72895488 512 159.72895488ZM512 223.72895488C441.307552 223.72895488 384 281.03650688 384 351.72895488 384 422.42140224 441.307552 479.72895488 512 479.72895488 582.692448 479.72895488 640 422.42140224 640 351.72895488 640 281.03650688 582.692448 223.72895488 512 223.72895488Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
        
        
      </div>
      
      
    </div>
  </div>
</div>
`;

exports[`Avatar groupMax demo works fine 1`] = `
<div
  class="demo-avatar"
>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar-group t-avatar--offset-right"
    >
      
      
      
      <div
        class="t-avatar t-avatar--circle"
      >
        <img
          alt=""
          src="https://tdesign.gtimg.com/site/avatar.jpg"
        />
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          Avatar
          
        </span>
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          +1
          
        </span>
      </div>
      
      
      
    </div>
  </div>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar-group t-avatar--offset-right"
    >
      
      
      
      <div
        class="t-avatar t-avatar--circle"
      >
        <img
          alt=""
          src="https://tdesign.gtimg.com/site/avatar.jpg"
        />
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          Avatar
          
        </span>
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        
        <svg
          class="t-icon t-icon-ellipsis"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 448C227.346224 448 256 476.653776 256 512 256 547.346224 227.346224 576 192 576 156.653776 576 128 547.346224 128 512 128 476.653776 156.653776 448 192 448ZM512 448C547.346224 448 576 476.653776 576 512 576 547.346224 547.346224 576 512 576 476.653776 576 448 547.346224 448 512 448 476.653776 476.653776 448 512 448ZM832 448C867.3462208 448 896 476.653776 896 512 896 547.346224 867.3462208 576 832 576 796.6537792 576 768 547.346224 768 512 768 476.653776 796.6537792 448 832 448Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
        <span>
          
          
          
        </span>
        
      </div>
      
      
      
    </div>
  </div>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar-group t-avatar--offset-right"
    >
      
      
      
      <div
        class="t-avatar t-avatar--circle"
      >
        <img
          alt=""
          src="https://tdesign.gtimg.com/site/avatar.jpg"
        />
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          Avatar
          
        </span>
      </div>
      <div
        class="t-avatar t-avatar--circle"
      >
        <span>
          
          
          
        </span>
      </div>
      
      
      
    </div>
  </div>
</div>
`;

exports[`Avatar shape demo works fine 1`] = `
<div
  class="demo-avatar"
>
  <div
    class="t-avatar t-avatar--circle"
  >
    <span>
      
      W
      
    </span>
  </div>
  <div
    class="t-avatar t-avatar--round"
  >
    <span>
      
       W 
      
    </span>
  </div>
</div>
`;

exports[`Avatar size demo works fine 1`] = `
<div
  class="demo-avatar"
>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar t-avatar--circle"
    >
      <span>
        
         W 
        
      </span>
    </div>
    <div
      class="t-avatar t-avatar--circle"
    >
      <span>
        
         W 
        
      </span>
    </div>
    <div
      class="t-avatar t-avatar--circle"
    >
      <span>
        
         W 
        
      </span>
    </div>
    <div
      class="t-avatar t-avatar--circle"
    >
      <span>
        
         W 
        
      </span>
    </div>
  </div>
  <div
    class="demo-avatar-block"
  >
    <div
      class="t-avatar t-avatar--round"
    >
      <span>
        
         W 
        
      </span>
    </div>
    <div
      class="t-avatar t-avatar--round"
    >
      <span>
        
         W 
        
      </span>
    </div>
    <div
      class="t-avatar t-avatar--round"
    >
      <span>
        
         W 
        
      </span>
    </div>
    <div
      class="t-avatar t-avatar--round"
    >
      <span>
        
         W 
        
      </span>
    </div>
  </div>
</div>
`;

exports[`Avatar size demo works fine 2`] = `
<div
  class="demo-avatar"
>
  <div
    class="t-avatar t-avatar--circle"
  >
    <span>
      
      王
      
    </span>
  </div>
  <div
    class="t-avatar t-avatar--circle"
  >
    <span>
      
      王亿
      
    </span>
  </div>
  <div
    class="t-avatar t-avatar--circle"
  >
    <span>
      
      王亿亿
      
    </span>
  </div>
</div>
`;
